linux/src/async_socket_linux.c

```diff
@@ -109,6 +110,13 @@ static int thread_worker_func(void* parameter)
     {
         struct epoll_event events[MAX_EVENTS_NUM];
         int num_ready = epoll_wait(g_epoll, events, MAX_EVENTS_NUM, EVENTS_TIMEOUT_MS);
+        if (num_ready == -1)
+        {
+            char err_msg[128] = {0};
+            (void)strerror_r(errno, err_msg, 128);
```

Comment:
>strerror_r [](http://example.com/codeflow?start=18&length=10)

From `man strerror_r`:

`POSIX.1-2001  and  POSIX.1-2008 require that a successful call to strerror() or strerror_l() shall leave errno
       unchanged, and note that, since no function return value is reserved to indicate an error, an application that
       wishes  to  check  for  errors should initialize errno to zero before the call, and then check errno after the
       call.`

Looks like errno NEEDS to be initialized to 0 before epoll_wait.